test(agw): Add scaling tests for dhcp mode

wolfseb opened the pull request:
With @MoritzThomasHuebner 

## Summary

This PR adds two scaling tests with repeated attach-detach loops for the non-nat mode with DHCP.
The locking in the release logic for this mode was changed, otherwise the database was locked for too long, so the release ran into a timeout.

## Test Plan

Check that the new s1ap integration tests are green. (might need re-provisioning of the dev VM, and needs the traffic server running with the udhcpd setup)
- [x] test_attach_detach_with_non_nat_dhcp_multi_ue.py
- [x] test_attach_detach_with_non_nat_dhcp_multi_ue_looped.py

## Additional Information

- [ ] This change is backwards-breaking

Commit:
test(agw): add looped attach detach integ tests for not nat

Signed-off-by: Sebastian Wolf <sebastian.wolf@tngtech.com>

Commit:
test(agw): enable usage of looped non nat dhcp tests

Signed-off-by: Sebastian Wolf <sebastian.wolf@tngtech.com>

Commit:
chore(agw): format and linter changes

Signed-off-by: Sebastian Wolf <sebastian.wolf@tngtech.com>

Commit:
chore(agw): move dhcp integ test to precommit in bazel

Signed-off-by: Sebastian Wolf <sebastian.wolf@tngtech.com>

LKreutzer commented on lte/gateway/python/integ_tests/defs.mk:
```diff
@@ -251,6 +251,8 @@ s1aptests/test_s1_handover_cancel.py \
 s1aptests/test_s1_handover_failure.py \
 s1aptests/test_s1_handover_timer_expiry.py \
 s1aptests/test_attach_and_mme_restart_loop_detach_and_mme_restart_loop_multi_ue.py \
+s1aptests/test_attach_detach_with_non_nat_dhcp_multi_ue.py \
```
What is the reasoning for making them non-sanity tests? Could they be part of the sanity tests or is the setup wrong?

mpfirrmann replied:
I would have the same question: Why are these nonsanity tests?

wolfseb replied:
The setup doesn't matter, it has its own setup steps within the test and it doesn't need anything else. We could put the in the sanity tests if you think it is more suitable, I don't have any strong opinion on where to put them.

mpfirrmann replied:
Well, sanity tests are checked in the CI workflows, while the non-sanity tests are just lying around in the code base. Therefore I would prefer to move them out of the nonsanity checks to be tested regularly.

LKreutzer commented on lte/gateway/python/integ_tests/s1aptests/BUILD.bazel:
```diff
@@ -2213,11 +2222,20 @@ pytest_test(
+
+pytest_test(
+    name = "test_attach_detach_with_non_nat_dhcp_multi_ue_looped",
+    size = "large",
```
Do the selected test sizes make sense for these tests?
![Screenshot from 2022-12-20 13-25-07](https://user-images.githubusercontent.com/34488763/208666856-70634fb6-35d7-47e5-a74d-be9ce69424fb.png)

MoritzThomasHuebner replied:
Yes, this test can take over five minutes to run and the other one is also of similar duration.

mpfirrmann commented on lte/gateway/python/integ_tests/s1aptests/test_attach_detach_with_non_nat_dhcp_multi_ue_looped.py:
```diff
@@ -0,0 +1,87 @@
+        self._s1ap_wrapper.cleanup()
+
+    def test_attach_detach_with_non_nat_dhcp(self):
+        # return
```
Remove comment

wolfseb replied:
Done.

mpfirrmann commented on lte/gateway/python/integ_tests/s1aptests/test_attach_detach_with_non_nat_dhcp_multi_ue_looped.py:
```diff
@@ -0,0 +1,87 @@
+    def test_attach_detach_with_non_nat_dhcp(self):
+        # return
+        """ Basic attach/detach test with two UEs and DHCP"""
+        num_ues = 64
```
Where does the number of UEs tested here come from? It 64 here and 32 in the other test.

wolfseb replied:
That is a leftover from testing, 32 should be enough. Thanks for spotting it! I changed it back to 32.

Commit:
chore(agw): change to 32 UEs for looped dhcp integ test

Also adapted the doc strings of the longer dhcp integ tests to better
describe what the tests are doing.

Signed-off-by: Sebastian Wolf <sebastian.wolf@tngtech.com>

Commit:
chore(agw): move long dhcp integ tests to sanity tests

Signed-off-by: Sebastian Wolf <sebastian.wolf@tngtech.com>

Commit:
fix(agw): do not wait for response when releasing dhcp ip

Signed-off-by: Sebastian Wolf <sebastian.wolf@tngtech.com>

MoritzThomasHuebner reviewed (approved):
LGTM

Tests are green locally on a branch that I rebased on master.
